js/popupView.js

```diff
@@ -0,0 +1,145 @@
+
+        open:function() {
+            if (this.isOpen) return;
+
+            this.isOpen = true;
+
+            Adapt.trigger('notify:triggerCustomView', this);
+        },
+
+        addItemClasses:function() {
+            var classes = this.model.get("_items")[this.currentIndex]._classes 
+                ? this.model.get("_items")[this.currentIndex]._classes
+                : '';  // _classes has not been defined
+      
+            this.$el.filter('.notify-popup').addClass('item-' + this.currentIndex + ' ' + classes);
+        },
+
+        removeItemClasses:function() {
+            var classes = this.model.get("_items")[this.currentIndex]._classes 
+                ? this.model.get("_items")[this.currentIndex]._classes
+                : '';  // _classes has not been defined
+      
+            this.$el.filter('.notify-popup').removeClass('item-' + this.currentIndex + ' ' + classes);
+        },
+
+        updateNavigation: function (index) {
+            var $nav = this.$('.hotgraphic-popup-nav'),
+                itemCount = this.model.get('_items').length;
+
+            $nav.removeClass('first').removeClass('last');
+
+            if(index <= 0 && !this.model.get('_canCycleThroughPagination')) {
+                this.$('.hotgraphic-popup-nav').addClass('first');
+                this.$('.hotgraphic-popup-controls.back').a11y_cntrl_enabled(false);
+                this.$('.hotgraphic-popup-controls.next').a11y_cntrl_enabled(true);
+            } else if (index >= itemCount-1 && !this.model.get('_canCycleThroughPagination')) {
+                this.$('.hotgraphic-popup-nav').addClass('last');
+                this.$('.hotgraphic-popup-controls.back').a11y_cntrl_enabled(true);
+                this.$('.hotgraphic-popup-controls.next').a11y_cntrl_enabled(false);
+            } else {
+                this.$('.hotgraphic-popup-controls.back').a11y_cntrl_enabled(true);
+                this.$('.hotgraphic-popup-controls.next').a11y_cntrl_enabled(true);
+            }
+            
+        },
+
+        onPopupOpened:function() {
+            this.$('.hotgraphic-item').hide().removeClass('active');
+            this.$('.hotgraphic-item').eq(this.currentIndex).show().addClass('active');
+            this.$('.hotgraphic-popup-count .current').html(this.currentIndex + 1);
+            this.$('.hotgraphic-popup-count .total').html(this.$('.hotgraphic-item').length);
+            this.$('.hotgraphic-popup').attr('class', 'hotgraphic-popup item-' + this.currentIndex);
+
+            this.$('.hotgraphic-popup-inner .active').a11y_focus();
+
+            this.addItemClasses();
+
+            this.updateNavigation(this.currentIndex);
+
+            this.trigger('hotGraphicPopup:opened');
+        },
+
+        onPopupClosed:function() {
+            this.remove();
+            this.trigger('hotGraphicPopup:closed');
+        },
+
+        previousHotGraphic:function(event) {
+            var targetIndex;
+
+            event.preventDefault();
```

Comment:
I think it's good to have an `if(event)` check here so you're not making the assumption that this function will only ever been called as a result of an event